lib/Sema/ConstraintSystem.cpp

```diff
@@ -1278,10 +1276,10 @@ Type ConstraintSystem::openPackExpansionType(PackExpansionType *expansion,
 void ConstraintSystem::recordOpenedPackExpansionType(PackExpansionType *expansion,
                                                      TypeVariableType *expansionVar) {
   bool inserted = OpenedPackExpansionTypes.insert({expansion, expansionVar}).second;
-  if (inserted) {
-    if (isRecordingChanges())
-      recordChange(SolverTrail::Change::recordedOpenedPackExpansionType(expansion));
-  }
+  ASSERT(inserted);
+
+  if (solverState)
```

Comment:
Could you remind me in what cases we don’t have a state available, I thought that it usually gets constructed very early, maybe some typeCheck* methods don’t have it initialized which might be a bug?…

Reply:
I've seen a few code paths where solverState was null, from CSGen I think.